index.js

```diff
@@ -320,6 +326,93 @@ module.exports = {
     return options;
   },
 
+  _shouldHandleTypeScript() {
+    if (this._cachedShouldHandleTypescript === undefined) {
+      // TODO: this doesn't account for things like in-repo addons and other
+      // weird inclusion structures
+      let checker = new VersionChecker(this.parent).for('ember-cli-typescript', 'npm');
```

Comment:
Instead of using `ember-cli-version-checker` here, I think it would probably be better (and may remove that TODO above) to use:

```js
let typescriptAddon = this.parent.addons.find(a => a.name === 'ember-cli-typescript');
return typescriptAddon !== undefined && semver.gte(typescriptAddon.pkg.version, '4.0.0-alpha.1');
```

Reply:
I'll give it a try!

Reply:
This worked well and had the benefit of being easier to write tests for.